### PR TITLE
Add tests and docs for Robocop API

### LINKED ISSUE #168: Add client-server mode with API
Client-server mode with API could be used by IDEs or separate programs for file scanning "on the fly" without need to wrap Robocop. 

Example:
```
robocop = Robocop.server(*args)
robocop.scan_file(file)
robocop.scan_file(file)
...
robocop.scan_file(file)
```

Returned output would need to be different than std.out (for example json object) so it would be handled by other tools.

### PR DESCRIPTION
**Pull request by bhirsz**
Closes #168

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
+.. _api:
```
Comment (bhirsz) [inline docs/api.rst:1]: Far from perfect but we can start with it - there are basic examples that should be sufficient.

File: robocop/utils/utils.py [hunk 2]
```diff
@@ -69,11 +69,11 @@ def issues_to_lsp_diagnostic(issues):
     return [{
         'range': {
             'start': {
-                'line': issue.line,
+                'line': issue.line - 1,
```
Comment (fabioz) [inline robocop/utils/utils.py:72]: This needs to be `'line': issue.line - 1 if issue.line > 0 else 0` (it seems there are some cases where 0 is reported and it became negative... in particular the: `Missing documentation in suite`).

File: docs/api.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    robocop_runner.reload_config()
+
+
+Scan files supplied in configuration
+------------------------------------
+To scan all files provided in ``Config`` invoke::
+
+    robocop_runner.run()
+
+``run()`` method by default return issues in JSON format.
```
Comment (mnojek) [inline docs/api.rst:37]:
```suggestion
``run()`` method by default returns issues in JSON format.
```

File: docs/api.rst [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+Scan existing robot ast model
+-----------------------------
+If you have ast model loaded before you can use ``robocop`` to scan it::
```
Comment (mnojek) [inline docs/api.rst:41]:
```suggestion
If you have ast model loaded before, you can use ``robocop`` to scan it::
```

**Pull request merged**